lib/hpcbase.pm

```diff
@@ -18,7 +18,8 @@ sub upload_service_log {
 }
 
 sub post_fail_hook {
-    select_virtio_console(force => 1);
+    my ($self) = @_;
+    $self->select_serial_terminal;
```

Comment:
The `force` argument is missing in `select_serial_terminal`, but when I understand  https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/69f94ece9d88a3086312b3a54a93433f564dc81a#diff-202b6b62531ad5ac248280ffc8c75598 correctly, I think it's the wonted behavior to fallback explicitly to virtio_console cause the other hang.
Maybe @asmorodskyi can say more.

Reply:
@cfconrad @asmorodskyi I agreed to add `force` argument, but now I think it was a mistake. As it was redundant to `VIRTIO_CONSOLE=1` setting, which enables it (why not just to set it?). Current solution enables virtio terminal by default, so it's not needed (variable `NO_VIRTIO_CONSOLE=1` is available for these who don't want to use virtio terminal).

The only problem is that until https://github.com/os-autoinst/os-autoinst/pull/1045 is merged and deployed, all tests should have `VIRTIO_CONSOLE=1`. Only HPC tests were using force, I'm willing to do the work (add them). OK Anton?

Reply:
the scenario which force me to add `force` parameter : 
I have some tests in HPC which incompatible with VIRTIO_CONSOLE=1 ( they switching to some other consoles so openQA not capable to properly detect this switch so only choice for me is screenshots ) but when they fail I want to fallback to virtio_console to collect some logs. Of course I can do it via `set_var` call but I just find the way with force parameter cleaner. And what is your thoughts against force parameter ?

Reply:
I thought the problem is, that some command which was executed in one of the shell's didn't terminate. So uploading logs will fail, cause we simply unable to run curl.
I faced a similar problem and I killed the command like this https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/run_ltp.pm#L84
But I think we will need something better, like forcing a relogin.

Reply:
didn't get why if command blocked in one shell it should block execution in other shells ?

Reply:
If test cannot use virtio terminal for some parts (which I explained bellow), it shouldn't have used `select_serial_terminal()` and just stay with plain `select_console()`. IMHO it'd be great to change test to be able to use virtio terminal and things would be solved.
What I have found the problem was that some HPC tests uses `send_key` and `assert_screen`.
I fixed that by calling `select_console('root-console')` in these cases.
Is that ok for you?
I'll setup openQA with TAP to test it.

Reply:
@pevik after reading PR further , I get your point ... yes we can drop this force parameter

Reply:
> didn't get why if command blocked in one shell it should block execution in other shells ?
If I'm not wrong, each `select_console('virtio-terminal')` retrieves the same serial terminal.

Anyway, as @asmorodskyi  explained, this is not the case here, so we can ignore my comment.